public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx

```diff
@@ -34,21 +35,29 @@ export const useGetGroupOptionsFromFolder = (folderTitle: string) => {
   const grafanaFolders = useCombinedRuleNamespaces(GRAFANA_RULES_SOURCE_NAME);
   const folderGroups = grafanaFolders.find((f) => f.name === folderTitle)?.groups ?? [];
 
-  const nonProvisionedGroups = folderGroups.filter((g) => {
-    return g.rules.every(
-      (r) => isGrafanaRulerRule(r.rulerRule) && Boolean(r.rulerRule.grafana_alert.provenance) === false
+  // we include provisioned folders, but disable the option to select them
+  const isProvisionedGroup = (group: CombinedRuleGroup) => {
```

Comment:
Should this be wrapped in `useCallback`?

Reply:
Usually I don't bother unless it's really necessary as a performance optimizations (when the components is re-rendered often).

Good shout-out here though since I believe this function doesn't even need to be in the hook in the first place – I'll move it outside.